packages/protoc-gen-es/src/transpile.ts

```diff
@@ -0,0 +1,189 @@
+import type { TSFile } from "@bufbuild/protoplugin/ecmascript";
+import { createTSFile } from "@bufbuild/protoplugin/ecmascript";
+import * as path from "path";
+import * as ts from "typescript";
+
+function createTranspiler(
+  options: ts.CompilerOptions,
+  files: TSFile[]
+): ts.Program {
+  const original = ts.createCompilerHost(options, true);
+  const virtualHost = new VirtualCompilerHost(original, files);
+
+  // Get a list of file names needed to create the program
+  const fileNames = files.map((file) => {
+    return file.name;
+  });
+
+  return ts.createProgram(fileNames, options, virtualHost);
+}
+
+class VirtualCompilerHost implements ts.CompilerHost {
+  private readonly _sourceFiles = new Map<TSFile, ts.SourceFile>();
+  private readonly _files = new Map<string, TSFile>();
+  private readonly _dirs = new Set<string>();
+
+  constructor(
+    private readonly wrapped: ts.CompilerHost,
+    files: readonly TSFile[]
+  ) {
+    for (let vf of files) {
+      // create map from path to file
+      if (this._files.has(vf.name)) {
+        throw new Error("Duplicate file paths in virtual files: " + vf.name);
+      }
+      this._files.set(vf.name, vf);
+      // create set of directory paths
+      let path = vf.name.split("/");
+      while (path.length > 1) {
+        path.pop();
+        this._dirs.add(path.join("/"));
+      }
+    }
+  }
+
+  lookupVirtualFile(fileName: string): TSFile | undefined {
+    let vf = this._files.get(fileName);
+    if (vf) return vf;
+    let cwd = process.cwd();
+    if (fileName.startsWith(cwd)) {
+      let relativePath = path.relative(cwd, fileName);
```

Comment:
I suspect that we can use `getCurrentDirectory()` and/or `getCanonicalFileName()` to avoid `process.cwd()`. Can you look into that? It seems wrong to need the cwd here.